checkout_and_build_pr.sh

```diff
@@ -0,0 +1,242 @@
+            REPO_CMD_HISTORY+="\033[0;33m     INFO: running [${CMD_TO_EXECUTE}], exit code ${EXIT_CODE} \n\033[0m"
+        fi
+    fi
+}
+
+
+function checkout_repo_branch() {
+    REPO_NAME=$1
+    BRANCH_NAME_TO_CHECKOUT=$2
+    GIT_SSH="git@github.com:ONSdigital/${REPO_NAME}.git"
+
+    REPO_CMD_HISTORY+="REPO: ${REPO_NAME} \n"
+
+    # echo "Getting cloning ${GIT_SSH}"
+    execute_and_record_command "git clone ${GIT_SSH} " true
+    pushd $REPO_NAME
+
+    execute_and_record_command "git checkout ${BRANCH_NAME_TO_CHECKOUT}" false
+}
+
+function checkout_and_build_repo_branch() {
+    REPO_NAME=$1
+    BRANCH_NAME_TO_CHECKOUT=$2
+    RUN_TESTS_CMD=$3
+    SKIP_TESTS_CMD=$4
+
+    checkout_repo_branch $REPO_NAME $BRANCH_NAME_TO_CHECKOUT
+
+    if [ "$SKIP_TESTS" = true ] ; then
+        echo "${REPO_NAME} Skipping Tests: Running CMD: ${SKIP_TESTS_CMD}"
+        execute_and_record_command "${SKIP_TESTS_CMD}" true        
+    else
+        echo "${REPO_NAME} Running Tests: Running CMD: ${RUN_TESTS_CMD}"
+        execute_and_record_command "${RUN_TESTS_CMD}" true  
+    fi
+
+    popd
+}
+
+function killOffRunningDocker() {
+
+    # Not sure if you'd never want to do this?  Doesn't cost any time if already clear
+    if [ "$KILL_DOCKER" = false ] ; then
+        echo "Leaving Docker alone"
+    else
+        echo 'Stopping any running Docker containers'
+        docker stop $(docker ps -aq)
+
+        echo 'Removing any stopped Docker containers'
+        docker rm $(docker ps -a -q)
+    fi
+
+}
+
+function createNewBaseDir() {
+    BRANCH_DIR_TO_MAKE=$1
+    BASE_DIR=$2
+
+    echo "Passed branch dir to make ${BRANCH_DIR_TO_MAKE}"
+
+    BASE_DIR="${BASE_DIR}/${BRANCH_DIR_TO_MAKE}"
+
+    # TODO:  What to do if dir alerady exists? rm it, fail etc.. 
+
+    echo "Making new base dir ${BASE_DIR}"
+    mkdir $BASE_DIR
+
+    cd $BASE_DIR
+    echo "Now in new DIR: ${PWD}"
+}
+
+function wait_until_containers_are_running_or_timeout() {
+    WAIT_FOR_DOCKER_UP_TIMEOUT_SECONDS=60
+    starting_containers=""
+
+    while true
+    do
+        starting_containers = $("docker ps | grep -E 'starting|unhealthy'")
+        echo -e "${starting_containers}"
+
+        if ["$starting_containers" = ""] ; then
+            echo "Looks like containers are up"
+            return
+        fi
+            
+        # Wait and try again in a second
+        sleep 1
+
+        echo "RES: ${starting_containers}"
+    
+        i=$[$i+1]
+    done
+
+    echo -e "\033[1;31m  ERROR: Docker Images did not come up in expected time \n ${starting_containers} \033[0m"
+    exit 1
+}
+
+
+########################################################################################################################
+#
+#         START OF SCRIPT
+#
+########################################################################################################################
+
+# Internal Variable, will use to record time
+SECONDS=0
+
+docker ps | grep -E 'starting|unhealthy'
+
+
+exit 1
```

Comment:
errr... your script is to exit every time, having done nothing? Not sure that's what we want